### DISCUSSION TITLE
Why solid-start instead of adding framework support for Remix ?

### DISCUSSION THREAD

**Discussion by asvny** [solidjs/solid-start#163] [General]
Honestly, I think `remix` has a great community and given that they have extracted the router implementation from `react-router` and this has paved way for other frameworks to add support Remix as a meta framework.

- Vue https://github.com/brophdawg11/remix-routers/blob/main/packages/vue/README.md
- Svelte https://github.com/brophdawg11/remix-routers/blob/svelte/packages/svelte/src/index.ts
- Angular https://dev.to/brandontroberts/remix-ing-routing-in-angular-4g90

Just curious to know ... why maintain a separate framework instead of adding support to an existing one ? I mean creating one is a LOT of work.
- Implementation
- Docs
- Plugins
- Community

**Comment by ryansolid** [comment]
You are correct this is a monumental effort and I often think that if Remix was around when I started(Mar 2021) I would have just done that. (By around I mean open sourced). I wouldn't be opposed to a Remix integration and have talked to Michael and Jacob a few times. At each point it is very clear the focus was on more popular libraries and there was concern in bringing in the custom transformations. That has changed in the last 2 months or so but a lot has changed in my perspective since starting the project.

1. Solid needs a better starting experience period.

Metaframeworks are just baseline. Especially for newer frameworks. Create React App doesn't do it anymore. Even with a Svelte Remix the core team is going to work on Sveltekit. For Solid, SolidStart represents the beginning of any project. CSR,SSR, and beyond.

2. We want the best experiences for Solid.

One of the challenges I've particularly had is with routing. Fine-grained reactivity does fine-grained updates of routing. We couldn't just apply the Remix patterns as they are course-grained. I had some talks with Michael about similarities of the core and while the matching could serve a good basis we'd need to change the way all updates were done, ie.. re-calling functions is a no-go.  That being said Solid already has a router with nested routing and parallelized data fetching. 

3. We want to leverage Solid best for the future

Solid's fine-grained rendering is actually pretty unique and Solid having access to what we are doing with core lets us move with core direction.  I think Remix has done a lot of great things but you can tell where they weren't lock step with core of React.  They implemented their own transition, and async server rendering mechanisms when React 18 was going to ship with them. Yes in so they've allowed a more agnostic core but it's trickier to leverage the benefits of specific framework. Remix will support React Server Components but React is building on a different model with them.

For Solid there are a lot of innovations that I'm happy to have explored with SolidStart. We've been doing streaming for over a year, we have incredible error boundaries that throw from server to client. We have kept the power of Remix's Loader/Actions but decoupled them from the URL. Typesafe RPCs as a foundation. I said I would have been happy with Remix a year ago and that's true, and still pull in a lot of influence. But it is becoming its own thing, something like Solid itself which will be standout. Just need some more time. The rest will come.
